contrib/src/main/scala/akka/stream/contrib/Valve.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.NotUsed
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler}
+import akka.stream._
+import akka.stream.contrib.SwitchMode.{Close, Open}
+
+import scala.concurrent.{Future, Promise}
+
+/**
+  * Pause/ Resume a Flow
+  */
+trait ValveSwitch {
```

Comment:
Can be made sealed as we do not expect any other implementations than our private one